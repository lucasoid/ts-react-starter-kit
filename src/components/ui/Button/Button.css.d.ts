export const btn: string;
export const btnPlaintext: string;
export const primary: string;
export const secondary: string;
