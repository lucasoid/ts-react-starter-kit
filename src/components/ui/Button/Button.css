.btn {
    padding: 0.5em 1em;
    box-shadow: none;
    border: none;
    cursor: pointer;
}

.primary {
    background-color: teal;
    color: #fff;
}

.secondary {
    background-color: #cecece;
    color: #444;
}

.btnPlaintext {
    padding: 0;
    box-shadow: none;
    background: none;
    border: none;
    color: inherit;
    cursor: pointer;
}
