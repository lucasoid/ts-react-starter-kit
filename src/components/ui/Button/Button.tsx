import * as React from 'react';
import * as styles from './Button.css';

interface IButtonProps {
    plaintext?: boolean;
    styles?: React.CSSProperties;
    onClick?: Function;
    children?: any;
}
export const Button: React.FC<IButtonProps> = props => (
    <button
        className={props.plaintext ? styles.btnPlaintext : styles.btn}
        style={{ ...props.styles }}
        onClick={evt => {
            evt.preventDefault();
            typeof props.onClick === 'function' && props.onClick();
        }}
    >
        {props.children}
    </button>
);
