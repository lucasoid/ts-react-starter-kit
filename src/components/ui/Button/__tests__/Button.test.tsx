import * as React from 'react';
import { shallow } from 'enzyme';
import { Button, ButtonTypes } from '../Button';

describe('Button', () => {
    it('renders a primary button', () => {
        let wrapper = shallow(<Button>Click here</Button>);
        expect(wrapper).toMatchSnapshot();
    });

    it('renders a secondary button', () => {
        let wrapper = shallow(<Button type={ButtonTypes.SECONDARY}>Click here</Button>);
        expect(wrapper).toMatchSnapshot();
    });

    it('renders a plaintext button', () => {
        let wrapper = shallow(<Button plaintext={true}>Click here</Button>);
        expect(wrapper).toMatchSnapshot();
    });

    it('renders the title attribute', () => {
        let wrapper = shallow(<Button title={'Alt text here'}>Click here</Button>);
        expect(wrapper).toMatchSnapshot();
    });

    it('accepts additional styles', () => {
        let wrapper = shallow(<Button styles={{ marginTop: '22px' }}>Click here</Button>);
        expect(wrapper).toMatchSnapshot();
    });

    it('invokes the onClick prop', () => {
        const onClick = jest.fn();
        let wrapper = shallow(<Button onClick={onClick}>Click here</Button>);
        wrapper.find('button').simulate('click', { preventDefault: () => {} });
        expect(onClick).toBeCalled();
    });

    it('gracefully handles a missing onClick prop', () => {
        const onClick = null;
        let wrapper = shallow(<Button onClick={onClick}>Click here</Button>);
        expect(() => wrapper.find('button').simulate('click', { preventDefault: () => {} })).not.toThrow();
    });
});
