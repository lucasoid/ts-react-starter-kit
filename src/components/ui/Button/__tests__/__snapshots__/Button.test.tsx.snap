// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button accepts additional styles 1`] = `
<button
  className="btn secondary"
  onClick={[Function]}
  style={
    Object {
      "marginTop": "22px",
    }
  }
>
  Click here
</button>
`;

exports[`Button renders a plaintext button 1`] = `
<button
  className="btnPlaintext secondary"
  onClick={[Function]}
  style={Object {}}
>
  Click here
</button>
`;

exports[`Button renders a primary button 1`] = `
<button
  className="btn secondary"
  onClick={[Function]}
  style={Object {}}
>
  Click here
</button>
`;

exports[`Button renders a secondary button 1`] = `
<button
  className="btn secondary"
  onClick={[Function]}
  style={Object {}}
>
  Click here
</button>
`;

exports[`Button renders the title attribute 1`] = `
<button
  className="btn secondary"
  onClick={[Function]}
  style={Object {}}
  title="Alt text here"
>
  Click here
</button>
`;
